### ISSUE TITLE
Add support for tab completion in the CLI

### ISSUE THREAD

**Issue by timbeiko** [hyperledger/besu#538]
Add support for tab completion wherever possible (for options, flags and subcommands) in the CLI. Default to PicoCLI’s built in behavior when possible.

*Acceptance Criteria:*
A user sees the options, flags and subcommands available to them when they begin typing and can tab through them to select the appropriate one.

**Comment by macfarla** [comment]
Teku implemented this so look at what they do

**Comment by diega** [comment]
@macfarla, are you referring to [this PR](https://github.com/ConsenSys/teku/pull/2234)?

**Comment by macfarla** [comment]
Yes I think so! Are you interested in working on this one @diega ?

**Comment by diega** [comment]
yes, I can take this

**Comment by macfarla** [comment]
Great - assigned to you @diega !
